lib/CL/clCreateBuffer.c

```diff
@@ -351,8 +351,239 @@ CL_API_ENTRY cl_mem CL_API_CALL POname (clCreateBuffer) (
 
   return mem;
 }
-POsym (clCreateBuffer)
+POsym (clCreateBuffer);
 
+// TBC: We probably want to limit tensor rank - should it be
+//      documented in the spec or queried at runtime?
```

Comment:
I think spec documentation should be enough. Is there a reasonable upper limit That Sould Be Good Enough for Everyone?